x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/textstructure/action/FindStructureAction.java

```diff
@@ -337,7 +373,16 @@ public ActionRequestValidationException validate() {
                         validationException
                     );
                 }
+                if (ECS_COMPATIBILITY_DISABLED.equalsIgnoreCase(ecsCompatibility) == false) {
+                    validationException = addIncompatibleArgErrorWithValue(
+                        ECS_COMPATIBILITY,
+                        TextStructure.Format.SEMI_STRUCTURED_TEXT,
+                        validationException,
+                        ecsCompatibility
+                    );
+                }
```

Comment:
I think the validation on the request should be changed.

`ecs_compatibility` should be validated to be either not specified, `disabled`, or `v1` regardless of what the structure is.  At the moment we'll happily let through an invalid value if semi-structured text has been specified, which will then throw an exception later after other work has been done.  And it's quite possible we'll want to use Grok patterns as part of the analysis of more structured formats, so people should be able to tell us to use ECS patterns for those.

(The reason the overall `grok_pattern` is disallowed is that it applies to the entire message, not subsections of the analysis, so that really only makes sense for semi-structured logs.)